### ISSUE TITLE
Do you use teacher Forcing during evaluation ?

### ISSUE THREAD

**Issue by agemagician** [google-research/text-to-text-transfer-transformer#56]
Hi,

During training, you use teacher forcing to accelerate the training and to have a better convergence but do you also use teacher forcing during the evaluation?
Can we rely on the evaluation results or do we need to use prediction and then calculate the performance of the model?

If you are using it during the evaluation, how we turn it off in the gin files?

Thanks in advance for the reply.

**Comment by craffel** [comment]
No, we don't use teacher forcing during evaluation (that would be cheating :) also - for some test sets we have no labels, so using teacher forcing would be impossible there). See the final sentence of the first paragraph in 3.1.2:

> At test time, we use greedy decoding (i.e. choosing the highest-probability logit at every timestep).

eval.gin will set utils.run.mode to "eval" which will result in the model sampling outputs autoregressively or, if beam_search.gin is also included, using beam search.
